cms/static/js/views/video/transcripts/utils.js

```diff
@@ -129,7 +130,10 @@ define(['jquery', 'underscore', 'jquery.ajaxQueue'], function($) {
                 match = link.pathname.match(/\/{1}([^\/]+)\.([^\/]+)$/);
                 if (match) {
                     cache[url] = {
-                        video: match[1],
+                        /* avoid too long video name, as it will be used as filename for video's transcript
+                        and a filename can not be more that 255 chars, limiting here to 100.
+                        */
+                        video: match[1].slice(0, maxLengthVideoName),
```

Comment:
this needs a method which can `slice` the matching string to max_length. We are slicing at two places, so it indicates to me that we can move the logic to a single place so that any change in required in future is done in that method.